consumer/pdata/metric.go

```diff
@@ -127,31 +127,41 @@ func (md Metrics) MetricAndDataPointCount() (metricCount int, dataPointCount int
 			ilm := ilms.At(j)
 			metrics := ilm.Metrics()
 			metricCount += metrics.Len()
-			ms := ilm.Metrics()
-			for k := 0; k < ms.Len(); k++ {
-				m := ms.At(k)
-				switch m.DataType() {
-				case MetricDataTypeIntGauge:
-					dataPointCount += m.IntGauge().DataPoints().Len()
-				case MetricDataTypeDoubleGauge:
-					dataPointCount += m.DoubleGauge().DataPoints().Len()
-				case MetricDataTypeIntSum:
-					dataPointCount += m.IntSum().DataPoints().Len()
-				case MetricDataTypeDoubleSum:
-					dataPointCount += m.DoubleSum().DataPoints().Len()
-				case MetricDataTypeIntHistogram:
-					dataPointCount += m.IntHistogram().DataPoints().Len()
-				case MetricDataTypeHistogram:
-					dataPointCount += m.Histogram().DataPoints().Len()
-				case MetricDataTypeSummary:
-					dataPointCount += m.Summary().DataPoints().Len()
-				}
-			}
+			dataPointCount += ilm.Metrics().DataPointCount()
 		}
 	}
 	return
 }
 
+// DataPointCount calculates the total number of  data points.
+func (ms MetricSlice) DataPointCount() (dataPointCount int) {
+	for k := 0; k < ms.Len(); k++ {
+		dataPointCount += ms.At(k).DataPointCount()
+	}
+	return
+}
+
+// DataPointCount calculates the total number of  data points.
+func (ms Metric) DataPointCount() (dataPointCount int) {
+	switch ms.DataType() {
+	case MetricDataTypeIntGauge:
+		dataPointCount = ms.IntGauge().DataPoints().Len()
+	case MetricDataTypeDoubleGauge:
+		dataPointCount = ms.DoubleGauge().DataPoints().Len()
+	case MetricDataTypeIntSum:
+		dataPointCount = ms.IntSum().DataPoints().Len()
+	case MetricDataTypeDoubleSum:
+		dataPointCount = ms.DoubleSum().DataPoints().Len()
+	case MetricDataTypeIntHistogram:
+		dataPointCount = ms.IntHistogram().DataPoints().Len()
+	case MetricDataTypeHistogram:
+		dataPointCount = ms.Histogram().DataPoints().Len()
+	case MetricDataTypeSummary:
+		dataPointCount = ms.Summary().DataPoints().Len()
+	}
+	return
+}
```

Comment:
Please avoid exposing public APIs unless strictly necessary and discussed in advance.

Reply:
I need this logic to split correctly.
The only other solution I can see is to move this logic to `splitmetrics.go` is this ok?

Reply:
implemented this proposition. PTAL